test/jdk/java/awt/Desktop/MailTest.java

```diff
@@ -103,6 +94,18 @@ private MailTest() {
 
     public static void main(String[] args) throws InterruptedException,
             InvocationTargetException {
+        if (!Desktop.isDesktopSupported()) {
+            String errorMessage = "Class java.awt.Desktop is not supported on " +
+                    "current platform. Further testing will not be performed";
+            throw new SkippedException(errorMessage);
+        }
+
+        Desktop desktop = Desktop.getDesktop();
+        if (!desktop.isSupported(Desktop.Action.MAIL)) {
+            String errorMessage = "Action.MAIL is not supported.";
+            throw new SkippedException(errorMessage);
```

Comment:
Since the error message is used only once, you can inline the message to the constructor.

You may inline `Desktop.getDesktop().isSupported(Desktop.Action.MAIL)` without introducing a local variable if you like.

Reply:
Done